app/static/js/base.js

```diff
@@ -5,45 +5,113 @@
             $card_body.append('<p class="card-text"> you found: </p><div class="d-flex"></div>');
             for (let resource in added_resource) {
-                $card_body.find('div').append('<div class="col text-center mt-2"></div>');
+                if (resource === 'diamond' && added_resource[resource].amount === 0)
+                    continue;
+                $card_body.find('div').append('<div class="col text-center mt-2 pl-0 pr-0"></div>');
                 $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource].picture}" /></p>`);
                 $card_body.find('.col').last().append(`<p class="card-text">${added_resource[resource].amount}</p>`);
             }
         }
 
+        function setPurchaseResultsMessage($purchaseResultsDiv, purchaseSuccess, text) {
+            $purchaseResultsDiv.removeClass('message-showing').empty();
+            $purchaseResultsDiv.append(`<p>${text}</p>`).addClass('message-showing');
+            if (purchaseSuccess)
+                $purchaseResultsDiv.find('p').addClass('bg-success');
+            else
+                $purchaseResultsDiv.find('p').addClass('bg-warning');
+            setTimeout(function() {
+                $purchaseResultsDiv.removeClass('message-showing');
+            }, 2000)
+        }
+
         function setupResourceSearch() {
             $('#search-resources-btn').on('click', function(){
                 let $btn = $(this),
-                    turns = $btn.data('turns-amount');
+                    $purchaseResult = $btn.closest('.card').find('.purchase-result'),
+                    turns = $userResources.data('army-turns'),
+                    purchaseSuccess = false;
 
-                if (turns < 10)
+                if (turns < 10) {
+                    setPurchaseResultsMessage($purchaseResult, purchaseSuccess, "You don't have enough turns");
                     return false;
+                }
 
                 $.gameApiGet('base/search_resources').done(function(user_new_resources){
-                    setUserResources($btn, user_new_resources, $userResources);
-                    setCardBody($btn, user_new_resources);
+                    setSpinner($btn, user_new_resources);
+                    setTimeout(function(){setUserResources(user_new_resources, $userResources)}, 3000);
+                    setTimeout(function(){setCardBody($btn, user_new_resources)}, 3000);
+                }).fail(function(status){
+                    if (status === 400)
+                        setPurchaseResultsMessage($purchaseResult, purchaseSuccess, "You don't have enough turns");
+                    else
+                        setPurchaseResultsMessage($purchaseResult, purchaseSuccess, "Status code");
+                })
+            });
+        }
+
+        function setupFieldSearch() {
+            $('#search-field-btn').on('click', function(){

```

Comment:
Try adding spaces between function declarations and `{`, looks cleaner
```suggestion
            $('#search-field-btn').on('click', function() {
```

Reply:
alright